### ISSUE TITLE
Gitea Workspace Service documentation

### ISSUE THREAD

**Issue by JaimieWi** [microsoft/AzureTRE#2801]
**Describe the bug**
The documentation for setting up Gitea as a Workspace Service is not very detailed and is out of date. 

I am trying to set up Gitea but there are steps within the docs that no longer apply. Such as: 

"The Gitea workspace service container image needs building and pushing:
`make build-gitea-workspace-service-image push-gitea-workspace-service-image`"

There is also not enough information or it is incorrect as following the steps gives a 403 error when trying to connect to the service.

**Steps to reproduce**

1. Run `make bundle-build bundle-publish DIR=templates/workspace_services/gitea`
1. Run `make bundle-register DIR=templates/workspace_services/gitea/ BUNDLE_TYPE=workspace_service`
1. Go to the TRE UI and Deploy Gitea
2. When Deploying fill in the required information:
2.1 Enter the OpenID Client ID is the Workspace App registration Application (client) ID. _This can be found in the KeyVault for the Workspace and on the Overview page of the App registration_
2.2 Enter the OpenID Client Secret. _This can only be found in the KeyVault for the Workspace._
2.3 Enter the OpenID Authority: https://login.microsoftonline.com/{tenant_id}/v2.0
3. Within the App registration, go to the Authentication tab in the blade on the left-hand side and add the Gitea redirect URI. In the form `https://<gitea_url>/user/oauth2/oidc/callback` (gitea_url can be found by clicking on the details tab of the deployed service)
4. Deploy the Workspace Service
**This is where it fails**
5. Press Connect on the Workspace Service. 
6. I see a 403 error 

<img width="788" alt="Screenshot 2022-10-31 at 16 01 40" src="https://user-images.githubusercontent.com/92854957/199053690-a0b5fcd3-063e-4814-8927-225ce47aa9fd.png">

In the documentation it says `Navigate to the Gitea workspace service and from the menu click the Sign in button.` Does this mean press Connect and it should ask you to sign in?

**Comment by marrobi** [comment]
Thanks @JaimieWi we will try and find time later this week to verify this.

cc @anastasiakuzn

**Comment by marrobi** [comment]
@JaimieWi  I've given the bundle a bit of an overhaul in PR #2862 

If you have chance to give this a go from my branch that would be great. I've tested and all seems good.

Let us know if any issues.

**Comment by JaimieWi** [comment]
Hi @marrobi, Thank you for looking at this! 

I have cloned your branch and tried to deploy the TRE but when running `make all` I see multiple errors similar to below.
![image](https://user-images.githubusercontent.com/92854957/201633045-549b8a20-4228-45ee-aa1d-9287f967b14c.png)
[tre_deploy_error.txt](https://github.com/microsoft/AzureTRE/files/10001887/tre_deploy_error.txt)

**Comment by marrobi** [comment]
What release of the TRE are you using? I believe this was fixed in #2761 , so at least v0.6.0.

**Comment by JaimieWi** [comment]
That is using the most up to date version of the TRE. I can try again using specifically v0.6.0 and see if it has a different outcome.

**Comment by marrobi** [comment]
@JaimieWi what TRE release are you running?

https://github.com/microsoft/AzureTRE/releases/tag/v0.6.0

**Comment by JaimieWi** [comment]
@marrobi I had no version specified, it was simply a clone of the latest commit to your branch. So should be v0.6.0.

**Comment by marrobi** [comment]
@JaimieWi - not my branch, the version of the main TRE deployment that you are deploying via CI/CD - as specified in https://github.com/microsoft/AzureTRE-Deployment/blob/eadf12b8ed4d9048e8b078080da2243659e73527/.devcontainer/devcontainer.json#L14

**Comment by JaimieWi** [comment]
That is currently pointing to a SHA for commit #2778 which is later than the release of v0.6.0. To test the Gitea service I was first deploying a test TRE but I will create a PR to do it via CI/CD now as well

**Comment by marrobi** [comment]
Hmm, wonder if your resource processor has updated, might be worth reimaging.

**Comment by JaimieWi** [comment]
Hi @marrobi, I managed to get it working in my TRE :) I'll update DEV and PROD TREs next but looks good, thank you!

**Comment by JaimieWi** [comment]
Hi @marrobi Apologies, I have now gone through the whole process and mirrored my own repository in Gitea. 

This initially failed with the Firewall showing `api.github.com` and `github.com` being denied in the firewall. I then added these to the Application Firewall rule related to gitea as Target FQDNs and it now works.
<img width="2049" alt="Screenshot 2022-11-15 at 13 49 13" src="https://user-images.githubusercontent.com/92854957/201935702-a246d205-778c-4afb-b2e5-c2e6102558c6.png">

**Comment by marrobi** [comment]
@JaimieWi I think that will provide access to the entire workspace subnet to those URLs s probably not a good solution.

If needing to mirror GitHub into the workspace then either go via a shared instance, or make sure the rule only relates to the Gitea compute, so the web app subnet (no users should ever be working in that subnet so is safe to allow).

What are you trying to achieve?

**Comment by JaimieWi** [comment]
@marrobi 

I did check and I cannot access `github.com` from within the VM. But to be safe, I have moved the rule over to the web_app_subnet and it works just as well.

The goal was so that researchers could have their private repositories in Gitea kept private and only accessible by logging in with their credentials. The shared service doesn't allow this as any repository has to be kept public to all within the TRE.

It would also be beneficial as researchers in a particular workspace may have a repository they would be happy to share within their project (workspace). But not across workspaces/projects.

Just to clarify, what is the purpose of the Gitea Workspace Service vs the Gitea Shared Service? 

There isn't anything in the documentation that helps us understand the distinction.
